pfrl/agents/hrl/hrl_controllers.py

```diff
@@ -46,29 +48,39 @@ def __init__(
         self.is_low_level = is_low_level
         self.minibatch_size = minibatch_size
         self.add_entropy = add_entropy
-        # create td3 agent
-        self.device = torch.device(f'cuda:{gpu}')
+
+        # create agent
         if self.add_entropy:
             def squashed_diagonal_gaussian_head(x):
-                mean, log_scale = torch.chunk(x, 2, dim=-1)
+                """
+                taken from the SAC code.
+                """
+                assert x.shape[-1] == action_dim * 2
+                mean, log_scale = torch.chunk(x, 2, dim=1)
                 log_scale = torch.clamp(log_scale, -20.0, 2.0)
                 var = torch.exp(log_scale * 2)
                 base_distribution = distributions.Independent(
                     distributions.Normal(loc=mean, scale=torch.sqrt(var)), 1
                 )
-                return base_distribution
+                # cache_size=1 is required for numerical stability
+                return distributions.transformed_distribution.TransformedDistribution(
+                    base_distribution, [distributions.transforms.TanhTransform(cache_size=1)]
+                )
 
+            # SAC policy definition:
             policy = nn.Sequential(
-                nn.Linear(state_dim + goal_dim, 300),
+                nn.Linear(state_dim + goal_dim, 256),
                 nn.ReLU(),
-                nn.Linear(300, 300),
+                nn.Linear(256, 256),
                 nn.ReLU(),
```

Comment:
Does this change of state size affect the result?

Reply:
I haven't verified that yet

Reply:
but it mainly follows SAC's no. of units.